test/AppNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,640 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "AppNotification-Test-Constants.h"
+#include "AppNotificationHelpers.h"
+#include "BaseTestSuite.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+using namespace AppNotificationHelpers;
+
+void BaseTestSuite::ClassSetup()
+{
+    ::Test::Bootstrap::SetupPackages();
+}
+
+void BaseTestSuite::ClassCleanup()
+{
+    ::Test::Bootstrap::CleanupPackages();
+}
+
+void BaseTestSuite::MethodSetup()
+{
+    ::Test::Bootstrap::SetupBootstrap();
+
+    bool isSelfContained{};
+    VERIFY_SUCCEEDED(TestData::TryGetValue(L"SelfContained", isSelfContained));
+
+    if (!isSelfContained)
+    {
+        ::WindowsAppRuntime::VersionInfo::TestInitialize(::Test::Bootstrap::TP::WindowsAppRuntimeFramework::c_PackageFamilyName);
+        VERIFY_IS_FALSE(::WindowsAppRuntime::SelfContained::IsSelfContained());
+    }
+    else
+    {
+        ::WindowsAppRuntime::VersionInfo::TestInitialize(L"I_don't_exist_package!");
```

Comment:
Why initialize this at all with a fake package?

Reply:
Here is the new guidelines for testing SelfContained: https://github.com/microsoft/WindowsAppSDK/blob/main/docs/Coding-Guidelines/SelfContained.md#unit-and-functional-testing

Reply:
But this should definitely be a test constant.

Reply:
Loved the package name 😁.